core/src/main/scala/filodb.core/store/ChunkSetInfo.scala

```diff
@@ -348,6 +350,15 @@ class CountingChunkInfoIterator(base: ChunkInfoIterator,
     // Why two counters ?
     // 1. query stats counter to meter usage by ws/ns. This will eventually be sent as part of query result.
     dataBytesScannedCtr.addAndGet(bytesRead)
+    if (dataBytesScannedCtr.get() > maxBytesScanned) {
```

Comment:
Do we need to add this check for every chunk read? Or can we reduce the frequency of this check for every range vector sent out? If latter, we could move this check into ExecPlan.scala

Reply:
On second thoughts, lets start with this. Group-by type queries can end up scanning more before the limits kick in with the other approach.